common/lib/xmodule/xmodule/assets/library_source_block/LibrarySourcedBlockPicker.jsx

```diff
@@ -0,0 +1,217 @@
+/* globals gettext */
+
+import 'whatwg-fetch';
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'underscore';
+import styles from './style.css';
+
+class LibrarySourcedBlockPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      libraries: [],
+      xblocks: [],
+      searchedLibrary: '',
+      libraryLoading: false,
+      xblocksLoading: false,
+      selectedLibrary: undefined,
+      selectedXblocks: new Set(this.props.selectedXblocks),
+    };
+    this.onLibrarySearchInput = this.onLibrarySearchInput.bind(this);
+    this.onXBlockSearchInput = this.onXBlockSearchInput.bind(this);
+    this.onLibrarySelected = this.onLibrarySelected.bind(this);
+    this.onXblockSelected = this.onXblockSelected.bind(this);
+    this.onDeleteClick = this.onDeleteClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchLibraries();
+  }
+
+  fetchLibraries(textSearch='', page=1, append=false) {
+    this.setState({
+      libraries: append ? this.state.libraries : [],
+      libraryLoading: true,
+    }, function() {
+      fetch(`/api/libraries/v2/?pagination=true&page=${page}&text_search=${textSearch}`)
+        .then(res => res.json())
```

Comment:
I'm also a bigger fan of async functions, for readability.  Did you by any chance have trouble with `await`, here?  If you just didn't try, no worries.  Consider this a non-blocking nit as well.